validator/keymanager/local/keymanager.go

```diff
@@ -276,44 +337,35 @@ func (km *Keymanager) CreateAccountsKeystore(
 			PublicKeys:  publicKeys,
 		}
 	} else {
-		existingPubKeys := make(map[string]bool)
-		existingPrivKeys := make(map[string]bool)
-		for i := 0; i < len(km.accountsStore.PrivateKeys); i++ {
-			existingPrivKeys[string(km.accountsStore.PrivateKeys[i])] = true
-			existingPubKeys[string(km.accountsStore.PublicKeys[i])] = true
-		}
-		// We append to the accounts store keys only
-		// if the private/secret key do not already exist, to prevent duplicates.
-		for i := 0; i < len(privateKeys); i++ {
-			sk := privateKeys[i]
-			pk := publicKeys[i]
-			_, privKeyExists := existingPrivKeys[string(sk)]
-			_, pubKeyExists := existingPubKeys[string(pk)]
-			if privKeyExists || pubKeyExists {
-				continue
-			}
-			km.accountsStore.PublicKeys = append(km.accountsStore.PublicKeys, pk)
-			km.accountsStore.PrivateKeys = append(km.accountsStore.PrivateKeys, sk)
-		}
+		updateAccountsStoreInMemory(km.accountsStore, privateKeys, publicKeys)
```

Comment:
You don't set `km.accountsStore` after executing this function call. Its better to be explicit of what struct fields are being updated in this method.

Reply:
updated the name to better represent this.